pyth_observer/check/stall_detection.py

```diff
@@ -0,0 +1,149 @@
+from dataclasses import dataclass
+from typing import List, Optional
+
+import numpy as np
+
+from pyth_observer.check.publisher import PriceUpdate
+
+
+@dataclass
+class StallDetectionResult:
+    """Results from stall detection analysis."""
+
+    is_stalled: bool
+    stall_type: Optional[
+        str
+    ]  # 'exact' for identical values, 'noisy' for artificial noise
+    base_price: Optional[float]
+    noise_magnitude: Optional[float]
+    duration: float  # how long the price has been stalled
+    confidence: float
+
+    @classmethod
+    def no_stall(cls) -> "StallDetectionResult":
+        """Create a StallDetectionResult instance indicating no stall detected."""
+        return cls(
+            is_stalled=False,
+            stall_type=None,
+            base_price=None,
+            noise_magnitude=None,
+            duration=0.0,
+            confidence=0.0,
```

Comment:
shouldn't the confidence be 1 in this case?

Reply:
true yep